storage/rocksdb/rdb_datadic.cc

```diff
@@ -925,13 +988,67 @@ uint Rdb_key_def::pack_record(const TABLE *const tbl, uchar *const pack_buffer,
     tuple = pack_field(field, &m_pack_info[i], tuple, packed_tuple, pack_buffer,
                        unpack_info, n_null_fields);
 
+    // Check if this column is covered.
+    if (unpack_info && m_index_type == INDEX_TYPE_SECONDARY) {
+      size_t data_length = field->data_length();
+      uint16 key_length;
+
+      // We always need the PK as a suffix, but in some cases MySQL won't add
+      // the PK suffix to a secondary key, and therefore it won't be included
+      // in key_info. If so, we need to get the key length for key parts from
+      // the PK directly.
+      DBUG_ASSERT(tbl->key_info != nullptr);
+      if (i >= tbl->key_info[get_keyno()].actual_key_parts) {
```

Comment:
Would this code run into a problem when actual_key_parts is greater than 16? The sk + pk parts could be greater than 16. Should it ignore values greater than 16? For example:

```
CREATE TABLE `t4` (
  `a` int(11) NOT NULL,
  `b` int(11) DEFAULT NULL,
  `c` int(11) DEFAULT NULL,
  `d` int(11) DEFAULT NULL,
  `e` int(11) DEFAULT NULL,
  `f` int(11) DEFAULT NULL,
  `g` int(11) DEFAULT NULL,
  `h` int(11) DEFAULT NULL,
  `i` int(11) DEFAULT NULL,
  `j` int(11) DEFAULT NULL,
  `k` int(11) DEFAULT NULL,
  `l` int(11) DEFAULT NULL,
  `m` int(11) DEFAULT NULL,
  `n` int(11) DEFAULT NULL,
  `o` int(11) DEFAULT NULL,
  `p` int(11) DEFAULT NULL,
  `q` int(11) DEFAULT NULL,
  `r` int(11) DEFAULT NULL,
  `s` int(11) NOT NULL,
  `t` int(11) NOT NULL,
  PRIMARY KEY (`a`,`s`,`t`),
  KEY `b` (`b`,`c`,`d`,`e`,`f`,`g`,`h`,`i`,`j`,`k`,`l`,`m`,`n`,`o`,`p`,`q`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 ;
```